### PR TITLE
157 Add Spectrum1DCollection.broaden

### PR DESCRIPTION
**Pull request by rebeccafair**
Addresses #157

This has also resulted in some refactoring of all Spectra.broaden methods. I've added some minimal tests for now checking that `Spectrum1DCollection.broaden` produces the same results as calling each `Spectrum1D.broaden` individually. I may add some other tests once the PDOS stuff is in and there are more options for test data.

### CODE REVIEW

File: euphonic/spectra.py [hunk 1]
```diff
@@ -644,6 +661,40 @@ def from_castep_phonon_dos(cls: SC, filename: str) -> SC:
...
+        -------
+        broadened_spectrum
+            A new Spectrum1DCollection object with broadened y_data
+
+        Raises
+        ------
+        ValueError
+            If shape is not one of the allowed strings
+        """
+        y_broadened = np.zeros(self._y_data.shape)
```
Comment (ajjackson) [inline euphonic/spectra.py:686]: A nice numpy shortcut for this (which also gets the dtype right) is `np.zeros_like(self._y_data)`. (Doesn't make much difference here, happy to leave as-is if you prefer.)

File: euphonic/spectra.py [hunk 2]
```diff
@@ -759,39 +810,14 @@ def broaden(self: S2D, x_width: Optional[Quantity] = None,
...
-                    self.get_bin_centres('y').magnitude,
-                    y_width.to(self.y_data_unit).magnitude,
-                    shape=shape)
-                z_broadened = correlate1d(
-                    z_broadened, y_broadening, mode='constant', axis=1)
-        else:
-            raise ValueError(
-                f"Distribution shape '{shape}' not recognised")
+        bin_centres = [self.get_bin_centres(ax).magnitude for ax in ['x', 'y']]
+        bwidths = [None]*2
```
Comment (ajjackson) [inline euphonic/spectra.py:814]: we do have space to write `bin_widths` here and it would be nicer to read

**Review by ajjackson — APPROVED** [review]
This looks good to me. Always nice to have some self-consistency tests in the mix!

File: euphonic/spectra.py (hunk 2 above)
Comment (rebeccafair) [reply euphonic/spectra.py]: Done

File: euphonic/spectra.py (hunk 1 above)
Comment (rebeccafair) [reply euphonic/spectra.py]: Done!

**Pull request merged**